### DISCUSSION TITLE
Context Menu not displaying correctly when displaying manually in a table from an observable subscribe block

### DISCUSSION THREAD

**Discussion by tripleentendre** [primefaces/community#363] [PrimeNG] [answered]
When trying to display a context menu when clicking a button in a table after making an http call the context method isn't displaying correctly. Please see [this stackblitz](https://stackblitz.com/edit/github-qw6mnk?file=src%2Fapp%2Fapp.component.html) that replicates the error. To make the issue occur click on any of the buttons in the table and then click another button in the table. 

![context_menu_issue](https://github.com/primefaces/community/assets/5608320/01553a7b-b430-4456-afa3-6c9ea7796c11)

If there's some work-around or something I've done incorrectly please let me know. 

Thank you!

**Comment by cetincakiroglu** [comment]
Hi,

In the example, I can manage to display all menus by button click. Your example doesn't replicate the issue. If you're referring to the code block below, contextmenu again displays correctly but somehow the async part, value assignment is blocking the `contextMenu.show()`, it works as it should when I remove that line. The solution is not about the contextmenu, it's the line above, maybe you can find another way to make it work.

```
  // Also seeing odd behavior when using async/await
  async openContextMenu(event: any, contextMenu: ContextMenu): Promise<void> {
      const value = await firstValueFrom(of(1));
      contextMenu.show(event);
  }
```

**Reply by tripleentendre** [reply]
Thank you for taking a look! I updated the description with more details on how to replicate the issue and also with a gif of what is happening.

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

You can use `appendTo` property of contextMenu. Here you can see in the example: https://stackblitz.com/edit/github-qw6mnk-53govf?file=src%2Fapp%2Fapp.component.html
